docusaurus/docs/Android/04-compose/07-guides/06-implementing-own-capabilities.mdx

```diff
@@ -0,0 +1,233 @@
+# Implementing Own Capabilities
+
+Each `Channel` has a list of capabilities that can be performed in it, such as muting a user, deleting messages and many
+others. Not all channels should have the same capabilities neither should all users, which is where own capabilities
+come into play.
+
+Inside of `Channel` you will find a property with a signature `ownCapabilities: Set<String>`. Which channel capabilities
+this set contains depends on the current user’s role (e.g. admin, guest, etc.), the channel type (e.g. messaging,
+livestream, etc.) and channel level settings. You can read more about channel
+capabilities [here](https://getstream.io/chat/docs/android/channel_capabilities/?language=kotlin) and more about channel
+level settings [here](https://getstream.io/chat/docs/android/channel-level_settings/).
+
+## Integrating channel capabilities into the UI
+
+If you are using our own UI in combination with our `ViewModel`s, then own capabilities work out of the box without any
+additional effort from your side, otherwise you will have to pass the correct set of capabilities to individual components.
+
+Different components have different ways of passing own capabilities to them, so we’ll go through them individually.
+
+:::note
+As a safety precaution, by default an empty set is passed wherever `ownCapabilities: Set<String>` is required. This gives the user no capabilities. Passing a set containing all capabilities will give the user potentially dangerous capabilities such as the ability to delete any message.
+:::
+
+### MessageComposer
+
+```kotlin
+@Composable
+public fun MessageComposer(
+        messageComposerState: MessageComposerState,
```

Comment:
This formatting is broken I think - should be 4 spaces of indentation

Reply:
I would also omit all the paramaters which A) don't contain capabilities, B) don't use them.

Otherwise htis snippet is just 40 lines of code where 10 lines are important

Reply:
FIxed indentation and omitted unnecessary parameters.